GradedModulePresentationsForCAP/gap/GradedModulePresentations.gi

```diff
@@ -466,25 +538,72 @@ InstallGlobalFunction( ADD_GRADED_KERNEL_RIGHT,
         
     end );
     
+    AddKernelLift( category,
+    
+    function( morphism, tau )
+        local underlying_lift;
+        
+      underlying_lift := KernelLift( UnderlyingPresentationMorphism( morphism ), UnderlyingPresentationMorphism( tau ) );
+        
+      return GradedPresentationMorphism( Source( tau ), underlying_lift, KernelObject( morphism ) );
+        
+    end );
+    
+end );
+
+##
+InstallGlobalFunction( ADD_GRADED_LIFT,
+                       
+  function( category )
+    
     AddLift( category,
       
-        ## TODO: Reference for the conventions for Lift
-#       function( alpha, beta )
-        function( beta, alpha )
+      function( alpha, beta )
         local lift;
         
-        lift := Lift( UnderlyingPresentationMorphism( beta ), UnderlyingPresentationMorphism( alpha ) );
+        lift := Lift( UnderlyingPresentationMorphism( alpha ), UnderlyingPresentationMorphism( beta ) );
         
         if lift = fail then
             return fail;
         fi;
         
-        return GradedPresentationMorphism( Source( beta ), lift, Source( alpha ) );
+        return GradedPresentationMorphism( Source( alpha ), lift, Source( beta ) );
         
     end );
     
 end );
 
+##
+InstallGlobalFunction( ADD_GRADED_COLIFT,
+                       
+  function( category )
+    
+    AddColift( category,
+      
+      function( alpha, beta )
+        local colift;
+        
+        colift := Colift( UnderlyingPresentationMorphism( alpha ), UnderlyingPresentationMorphism( beta ) );
```

Comment:
I already have a function "ReductionOfHomalgMatrix( mat, degree_mat )" that does the job. I will talk with Mohamed about adding it to the package "MatricesForHomalg", so that we can use it here to compute the degree zero part of the underlying "maybe" non-homogeneous lift/colift.

Reply:
Okay :). Then please make sure the right version of MatricesForHomalg is loaded, please.